app/controllers/cards_controller.rb

```diff
@@ -8,17 +8,25 @@ def new
   end
 
   def create
-    @card = Card.new(card_params.merge(user: current_user))
-
-    if @card.save
-      redirect_to cards_path
+    @deck = Deck.find_by(id: params[:card][:deck])
+    if !@deck
+      flash[:error] = "Deck can't be blank"
+      redirect_to new_card_path
     else
-      render 'new'
+      @card = Card.new(card_params.merge(user: current_user, deck: @deck))
+
+      if @card.save
+        redirect_to cards_path
+      else
+        render 'new'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.